gobblin-core/src/main/java/gobblin/publisher/HiveRegistrationPublisher.java

```diff
@@ -65,22 +66,48 @@
  * @author Ziyang Liu
  */
 @Slf4j
+@Alias("hivereg")
 public class HiveRegistrationPublisher extends DataPublisher {
 
   private static final String DATA_PUBLISH_TIME = HiveRegistrationPublisher.class.getName() + ".lastDataPublishTime";
   private static final Splitter LIST_SPLITTER_COMMA = Splitter.on(",").trimResults().omitEmptyStrings();
   public static final String HIVE_SPEC_COMPUTATION_TIMER = "hiveSpecComputationTimer";
+  private static final String PATH_DEDUPE_ENALED = "hive.registration.path.dedupe.enabled";
+  private static final boolean DEFAULT_PATH_DEDUPE_ENALED = true;
+
   private final Closer closer = Closer.create();
   private final HiveRegister hiveRegister;
   private final ExecutorService hivePolicyExecutor;
   private final MetricContext metricContext;
 
+  /**
+   * The configuration to determine if path deduplication should be enabled during Hive Registration process.
+   * Recall that HiveRegistration iterate thru. each topics' data folder and obtain schema from newest partition,
+   * it might be the case that a table corresponding to a registered path has a schema changed.
+   * In this case, path-deduplication won't work.
+   *
+   * e.g. In streaming mode, there could be cases that files(e.g. avro) under single topic folder carry different schema.
+   */
+  private boolean isDefaultPathDedupeEnaled;
```

Comment:
Can just be enablePathDedupe or isPathDedupeEnabled. The default in the name is confusing since it is not a default value.